google/cloud/aiplatform/datasets/datasources.py

```diff
@@ -0,0 +1,140 @@
+
+        if gcs_source:
+            dataset_metadata = {"input_config": {"gcs_source": {"uri": gcs_source}}}
+        elif bq_source:
+            dataset_metadata = {"input_config": {"bigquery_source": {"uri": bq_source}}}
+
+        self._dataset_metadata = dataset_metadata
+
+    @property
+    def dataset_metadata(self) -> Optional[Dict]:
+        """Dataset Metadata."""
+        return self._dataset_metadata
+
+
+class NonTabularDatasource(_Datasource):
+    """Datasource for creating a non-tabular dataset for AI Platform"""
+
+    def __init__(self):
+        """Creates an empty non-tabular datasource"""
+        self._dataset_metadata = {}
+
+    @property
+    def dataset_metadata(self) -> Optional[Dict]:
+        return self._dataset_metadata
+
+
+class NonTabularDatasourceImportable(NonTabularDatasource, _DatasourceImportable):
+    """Datasource for creating a non-tabular dataset for AI Platform and importing data to the dataset"""
+
+    def __init__(
+        self,
+        gcs_source: Sequence[str],
+        import_schema_uri: str,
+        data_item_labels: Optional[Dict] = {},
+    ):
+        """Creates a non-tabular datasource
+
+        Args:
+            gcs_source: Sequence[str]
+                Required. The Google Cloud Storage location for the input content.
+                Google Cloud Storage URI(-s) to the input file(s). May contain
+                wildcards. For more information on wildcards, see
+                https://cloud.google.com/storage/docs/gsutil/addlhelp/WildcardNames.
+            import_schema_uri: (str)
+                Required. Points to a YAML file stored on Google Cloud
+                Storage describing the import format. Validation will be
+                done against the schema. The schema is defined as an
+                `OpenAPI 3.0.2 Schema
+            data_item_labels: Optional[Dict] = {}
+                Labels that will be applied to newly imported DataItems. If
+                an identical DataItem as one being imported already exists
+                in the Dataset, then these labels will be appended to these
+                of the already existing one, and if labels with identical
+                key is imported before, the old label value will be
+                overwritten. If two DataItems are identical in the same
+                import data operation, the labels will be combined and if
+                key collision happens in this case, one of the values will
+                be picked randomly. Two DataItems are considered identical
+                if their content bytes are identical (e.g. image bytes or
+                pdf bytes). These labels will be overridden by Annotation
+                labels specified inside index file refenced by
+                ``import_schema_uri``,
+                e.g. jsonl file.
+        """
+        super().__init__()
+        self._gcs_source = utils.sequence_to_list(gcs_source)
```

Comment:
I had a discussion in another comment about using `Union[str, Sequence[str]]` instead of using `sequence_to_list`